internal/resources/cluster/integration/data_source_integration.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package integration
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func DataSourceIntegration() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIntegrationRead,
+		Schema:      integrationSchema,
+	}
+}
+
+func dataSourceIntegrationRead(_ context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	fn := constructFullName(d)
+
+	resp, err := config.TMCConnection.IntegrationResourceService.ManageV1alpha1ClusterIntegrationResourceServiceRead(fn)
+	if err != nil {
+		if clienterrors.IsNotFoundError(err) {
+			d.SetId("")
+			return
+		}
+
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control integration entry, name: %s", fn.Name))
+	}
+
+	d.SetId(resp.Integration.Meta.UID)
+
+	// FullName
+	if err = d.Set(managementClusterNameKey, resp.Integration.FullName.ManagementClusterName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set(provisionerNameKey, resp.Integration.FullName.ProvisionerName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set(clusterNameKey, resp.Integration.FullName.ClusterName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = d.Set(nameKey, resp.Integration.FullName.Name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Meta
+	if err = d.Set(common.MetaKey, common.FlattenMeta(resp.Integration.Meta)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Spec
+	if err = d.Set(configurationsKey, flattenIntegrationSpec(resp.Integration.Spec)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Status
```

Comment:
all the status need not be set. Just set a handful attributes which add values the resource.
refer to the cluster resource status